pkg/resourceinterpreter/defaultinterpreter/aggregatestatus.go

```diff
@@ -280,10 +281,9 @@ func aggregatePodStatus(object *unstructured.Unstructured, aggregatedStatusItems
 
 	newStatus := &corev1.PodStatus{}
 	newStatus.ContainerStatuses = make([]corev1.ContainerStatus, 0)
-	runningFlag := true
+	statusSet := sets.NewString()
 	for _, item := range aggregatedStatusItems {
 		if item.Status == nil {
-			runningFlag = false
 			continue
 		}
```

Comment:
```golang
		if item.Status == nil {
			// maybe pod's status hasn't been collected yet, assume it's in pending state.
			// Otherwise, it may affect the final aggregated state.
			podPhases.Insert(string(corev1.PodPending))
			continue
		}
```
We can't ignore the item with `nil` status.

Reply:
Please help cover this scenario in the unit test.

Reply:
Done!